File: samples-and-tests/yabe-with-scala/app/models/Comment.java
package models;
 
import java.util.*;
import javax.persistence.*;
 
import play.db.jpa.*;
import play.data.validation.*;
 
@Entity
public class Comment extends Model {
 
    @Required
    public String author;
    
    @Required
    public Date postedAt;
     
    @Lob
    @Required
    @MaxSize(10000)
    public String content;
    
    @ManyToOne
    @Required
    public Post post;
    
    public Comment(Post post, String author, String content) {
        this.post = post;
        this.author = author;
        this.content = content;
        this.postedAt = new Date();
    }
    
    public String toString() {
        return content.length() > 50 ? content.substring(0, 50) + "..." : content;
    }
 
}

File: samples-and-tests/yabe-with-scala/app/models/Comment.scala
package models
 
import java.util._
import javax.persistence._
 
import play.db.jpa._
import play.data.validation._
 
@Entity
class Comment(
       
       @ManyToOne
       @Required
       var post: Post,
       
       @Required
       var author: String, 
       
       @Lob
       @Required
       @MaxSize(10000)
       var content: String

) extends Model[Comment] {
    
    @Required
    var postedAt = new Date()
    
    override def toString() = {
        if(content.length() > 50) content.substring(0, 50) else content
    }
 
}
